src/coreComponents/mesh/CellElementRegionSelector.cpp

```diff
@@ -0,0 +1,265 @@
+  std::set< string > const requestedMatchPatterns = toStdSet(
+    region.getReference< string_array >( ViewKeys::cellBlockMatchPatternsString() ) );
+  std::set< string > const requestedCellBlock = toStdSet(
+    region.getReference< string_array >( ViewKeys::sourceCellBlockNamesString() ) );
+
+  // at least one selection method is needed
+  GEOS_THROW_IF( requestedAttributeValues.empty() && requestedMatchPatterns.empty() && requestedCellBlock.empty(),
+                 GEOS_FMT( "{}: cellBlocks must be selected to fill this region (using {}, {} or {}).",
+                           region.getDataContext(),
+                           ViewKeys::sourceCellBlockNamesString(),
+                           ViewKeys::cellBlockAttributeValuesString(),
+                           ViewKeys::cellBlockMatchPatternsString() ),
+                 InputError );
+  // only one selection method allowed for now
+  GEOS_THROW_IF( ( !requestedAttributeValues.empty() + !requestedMatchPatterns.empty() + !requestedCellBlock.empty() ) != 1,
+                 GEOS_FMT( "{}: Only one setting must be used to select cellBlocks ({}, {} or {}).",
+                           region.getDataContext(),
+                           ViewKeys::sourceCellBlockNamesString(),
+                           ViewKeys::cellBlockAttributeValuesString(),
+                           ViewKeys::cellBlockMatchPatternsString() ),
+                 InputError );
+
+  std::set< string > const matchPatterns = buildMatchPatterns( region, requestedAttributeValues, requestedMatchPatterns );
+  std::set< string > const matchedCellBlocks = getMatchingCellblocks( region, matchPatterns );
+  std::set< string > cellBlocksSelection = requestedCellBlock;
+  cellBlocksSelection.insert( matchedCellBlocks.begin(), matchedCellBlocks.end() );
+  verifyRequestedCellBlocks( region, cellBlocksSelection );
+
+  registerRegionSelection( region, requestedAttributeValues, cellBlocksSelection );
+
+  return cellBlocksSelection;
+}
+
+void CellElementRegionSelector::checkSelectionConsistency() const
+{
+  auto const getDataContextsStr = []( auto const & groupList ) -> string {
+    std::ostringstream oss;
+    for( Group const * group : groupList )
+    {
+      oss << GEOS_FMT( "- {}\n", group->getDataContext() );
+    }
+    return oss.str();
+  };
+
+  // Search of never or multiple selected attribute values
+  std::set< string > orphanRegionAttributes;
+  for( auto const & [attributeValueStr, owningRegions] : m_regionAttributeOwners )
+  {
+    if( owningRegions.size() == 0 )
+    {
+      orphanRegionAttributes.insert( attributeValueStr );
+    }
+    GEOS_THROW_IF( owningRegions.size() > 1,
+                   GEOS_FMT( "The region attribute '{}' has been referenced in multiple {}:\n- {}\n- {}",
+                             attributeValueStr, CellElementRegion::catalogName(),
+                             getDataContextsStr( owningRegions ) ),
+                   InputError );
```

Comment:
Would you like to wait for all the "errors" to be found before you log them all?

Reply:
Good suggestion!